pkg/dl/dl.go

```diff
@@ -36,16 +36,22 @@ const (
 	RTLD_DEEPBIND = C.RTLD_DEEPBIND
 )
 
+type Config struct {
+	Name  string
+	Flags int
+}
+
```

Comment:
Just curious, why was this level of indirection added? Having to call this feels awkward to me, for example: `nvml.Config.Name`.

I'd almost rather just see the global variable we have renamed to `nvmllib` and then leave this as is, so we get `nvmllib.Name`.

That said -- maybe `Name` isn't the right thing anymore given what we are trying to accomplish in this PR -- and `Path`  is probably better.

Reply:
Also, I have not yet changed the `dl` package. I will rather do that in a follow-up.